### ISSUE TITLE
[GPU] Softmax dispatch has two reductions and one of them which doesnt have a lowering config is not getting tiled

### ISSUE THREAD

**Issue by nirvedhmeshram** [iree-org/iree#19406]
Following [dispatch](https://gist.github.com/nirvedhmeshram/0562ed01cac2e6b2e47a572dfe1926cb#file-problem_disptach-mlir) was isolated to be generating a large private allocation in #19385 

The reason for this is that the dispatch presents with two reduction linalg generics as shown [here](https://gist.github.com/nirvedhmeshram/0562ed01cac2e6b2e47a572dfe1926cb#file-problem_dispatch_exec_config) but notice that we set config on only one of them. Manually adding a config to the other one fixes the allocation problem.

There are two possible solutions
1. Add config propagation logic so that both ops get the config
2. Add reduction tiling in `TileLargeTensors`

We probably should do both with 1. being the real solution and 2. being a fallback. 

Here is a compile script to generate the artifacts and full dump
```
#!/bin/bash
# compile full dump
iree-compile \
    "$(dirname $0)/problem_dispatch.mlir" \
    --iree-input-type=auto \
    --iree-vm-bytecode-module-output-format=flatbuffer-binary \
    "-o=$(dirname $0)/core-output.vmfb" \
    --mlir-print-ir-after-all \
    --iree-hal-target-device=hip \
    --iree-hip-target=gfx942 \
    &> output.mlir


# compile to exec configs
iree-compile \
    "$(dirname $0)/problem_dispatch.mlir" \
    --iree-input-type=auto \
    --iree-vm-bytecode-module-output-format=flatbuffer-binary \
    "-o=$(dirname $0)/problem_dispatch_exec_config.mlir" \
    --iree-hal-target-device=hip \
    --iree-hip-target=gfx942 \
    --compile-to=executable-configurations
```

**Comment by nirvedhmeshram** [comment]
cc @qedawkins @MaheshRavishankar

**Comment by MaheshRavishankar** [comment]
We've discussed this before, but "propagation" is a very unstable thing in general. Maybe we can try to find a way that is not propagation based.

**Comment by nirvedhmeshram** [comment]
> We've discussed this before, but "propagation" is a very unstable thing in general. Maybe we can try to find a way that is not propagation based.

That SGTM, uopdated issue title to reflect the actual problem.

**Comment by qedawkins** [comment]
Propagation might not be the right term here, but this is a case where we have two ops with unrelated reduction dimensions that both need to be tiled. This means we need to be able to set a configuration on both ops, which means the workgroup size, subgroup size, thread tile sizes, and workgroup tile sizes need to be consistent. Another option is to stop setting reduction/thread tile sizes in KernelConfig and leave support for pre-specified tile sizes only for tuning purposes.

**Comment by MaheshRavishankar** [comment]
> Propagation might not be the right term here, but this is a case where we have two ops with unrelated reduction dimensions that both need to be tiled. This means we need to be able to set a configuration on both ops, which means the workgroup size, subgroup size, thread tile sizes, and workgroup tile sizes need to be consistent. Another option is to stop setting reduction/thread tile sizes in KernelConfig and leave support for pre-specified tile sizes only for tuning purposes.

This makes sense. We can have multiple lowering configs, but only one of them can carry the workgroup size information (and that is also the root op that is used for distribution tiling). I am happy to help make that consistent

**Comment by qedawkins** [comment]
Yeah still single workgroup tile size per DAG, but the other sizes do need something a little more advanced.
